src/components/views/settings/Notifications.tsx

```diff
@@ -418,7 +419,8 @@ export default class Notifications extends React.PureComponent<IProps, IState> {
             }
             const kind = PushRuleKind.ContentSpecific;
             for (const word of diff.added) {
-                await MatrixClientPeg.get().addPushRule('global', kind, word, {
+                const newKey = encodeURIComponent(word.replace(/\./g, "%"));
```

Comment:
it feels relatively arbitrary to replace a broken character with a different character, then prevent people from using that new special character. Is there a naming structure we can come up with which won't break things, and is unlikely to appear in keywords today?

Reply:
Would  using &#46 be fine? The &#46 is the HTML coded character for periods(.).

I also found this %2E for periods(.). However, it doesn't work well with all browsers and it is automatically converted to periods(.) in some cases.

Reply:
A string prefix would be better than a replacement character.

Reply:
Would using a string prefix mean that the (.) would still be included in the key passed to the addPushRule function?
 
This means the (.) would still be parsed to the path, which would break the link afterwards.
<img width="633" alt="Screenshot 2022-04-14 at 23 52 06" src="https://user-images.githubusercontent.com/59070723/163490508-309e252b-8679-452d-bf42-b0d6b4f2de07.png">

Reply:
Could you clarify more on this? Thank you